### PR TITLE
Hardening manifests

### PR DESCRIPTION
**Pull request by LukasAud**
Prior to this commit some commands in our manifest classes would not be up to our code standards.

This commit aims to sanitise our code according to our teams best practices.

### CODE REVIEW

File: manifests/adopt.pp [hunk 1]
```diff
@@ -253,10 +253,10 @@
 
   case $_package_type {
```
Comment (ekohl) [inline manifests/adopt.pp:254]: Unrelated, but I'm reading through this and I'm wondering what's the difference between the two: default and `.tar.gz` look the same.

File: manifests/config.pp [hunk 2]
```diff
@@ -3,10 +3,13 @@
   case $facts['os']['family'] {
     'Debian': {
       if $java::use_java_alternative != undef and $java::use_java_alternative_path != undef {
+        $command_debian = ['update-java-alternatives', '--set', $java::use_java_alternative, $java::jre_flag]
+        $unless_debian = ["test /etc/alternatives/java -ef '${java::use_java_alternative_path}'"]
```
Comment (ekohl) [inline manifests/config.pp:7]: This still passes a raw string.
```suggestion
        $unless_debian = [['test', '/etc/alternatives/java', '-ef', $java::use_java_alternative_path]]
```

File: manifests/config.pp [hunk 3]
```diff
@@ -22,18 +25,23 @@
         # The standard packages install alternatives, custom packages do not
         # For the stanard packages java::params needs these added.
         if $java::use_java_package_name != $java::default_package_name {
+          $command_redhat = ['alternatives', '--install', '/usr/bin/java', 'java', $java::use_java_alternative_path, '20000']
+          $unless_redhat = ["alternatives --display java | grep -q ${java::use_java_alternative_path}"]
```
Comment (ekohl) [inline manifests/config.pp:29]: This relies on a shell, so perhaps explicitly keep it a string to make that clear?
```suggestion
          $unless_redhat = "alternatives --display java | grep -q ${java::use_java_alternative_path}"
```

Then you should escape the value.

File: manifests/config.pp [hunk 4]
```diff
@@ -22,18 +25,23 @@
...
             path    => '/usr/bin:/usr/sbin:/bin:/sbin',
-            command => "alternatives --install /usr/bin/java java ${$java::use_java_alternative_path} 20000" ,
-            unless  => "alternatives --display java | grep -q ${$java::use_java_alternative_path}",
+            command => $command_redhat,
+            unless  => $unless_redhat,
             before  => Exec['update-java-alternatives'],
           }
         }
+        $command_default = ['alternatives', '--set', 'java', $java::use_java_alternative_path]
+        $unless_default = ["test /etc/alternatives/java -ef '${java::use_java_alternative_path}'"]
```
Comment (ekohl) [inline manifests/config.pp:39]:
```suggestion
        $unless_default = [['test', '/etc/alternatives/java', '-ef', $java::use_java_alternative_path]]
```

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by LukasAud** [comment]
Thanks @ekohl for your suggestions. After carefully testing them, I'm happy enough to implement them into the PR. As for your question, I'm not really sure myself what the difference might be. Maybe someone more veteran, such as @david22swan might be able to help answer that.

File: manifests/adopt.pp (hunk 1 above)
Comment (ekohl) [reply manifests/adopt.pp:254]: To be explicit, I think this whole case statement can be replaced by `$install_command = ['tar', '-zxf', $destination, '-C', $_basedir]`.

Comment (LukasAud) [reply manifests/adopt.pp]: I briefly discussed the idea with another engineer. The conclusion was that the code is probably obsolete and we agreed to apply your suggestion. Thanks.

**Review by david22swan — APPROVED** [review]
LGTM

**Pull request merged**